internal/juju/applications.go

```diff
@@ -1280,7 +1280,7 @@ func (c applicationsClient) computeSetCharmConfig(
 	}
 
 	apiCharmID := apiapplication.CharmID{
-		URL:    resolvedURL,
+		URL:    newURL.String(),
```

Comment:
This needs to be resolvedURL.String() instead. It's causing the acceptance test failures.

Reply:
done